fulp_modules/QM_wrestling/code/wrestling.dm

```diff
@@ -0,0 +1,455 @@
+	to_chat(user, "<span class='danger'>You lift [target] up!</span>")
+	FlipAnimation()
+	for(var/i = 0, i < 3, i++)
+		if(user && target)
+			user.pixel_y += 3
+			target.pixel_y += 3
+			user.setDir(turn(user.dir, 90))
+			target.setDir(turn(target.dir, 90))
+			switch(user.dir)
+				if(NORTH)
+					target.pixel_x = user.pixel_x
+				if(SOUTH)
+					target.pixel_x = user.pixel_x
+				if(EAST)
+					target.pixel_x = user.pixel_x - 8
+				if(WEST)
+					target.pixel_x = user.pixel_x + 8
+			if(get_dist(user, target) > 1)
+				to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+				return
+			if(!isturf(user.loc) || !isturf(target.loc))
+				to_chat(user, "<span class='warning'>You can't slam [target] here!</span>")
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+				return
+		else
+			if(user)
+				user.pixel_x = user.base_pixel_x
+				user.pixel_y = user.base_pixel_y
+			if(target)
+				target.pixel_x = target.base_pixel_x
+				target.pixel_y = target.base_pixel_y
+			return
+		sleep(1)
+	if(user && target)
+		user.pixel_x = user.base_pixel_x
+		user.pixel_y = user.base_pixel_y
+		target.pixel_x = target.base_pixel_x
+		target.pixel_y = target.base_pixel_y
+
+		if(get_dist(user, target) > 1)
+			to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+			return
+		if(!isturf(user.loc) || !isturf(target.loc))
+			to_chat(user, "<span class='warning'>You can't slam [target] here!</span>")
+			return
+		target.forceMove(user.loc)
+		var/fluff = "body-slam"
+		target.visible_message("<span class='danger'>[user] [fluff] [target]!</span>", \
+						"<span class='userdanger'>You're [fluff]ed by [user]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, user)
+		to_chat(user, "<span class='danger'>You [fluff] [target]!</span>")
+		playsound(user.loc, "swing_hit", 50, TRUE)
+		if(!target.stat)
+			target.emote("scream")
+			target.Paralyze(30)
+			target.adjustBruteLoss(rand(10,15))
+		else
+			target.adjustBruteLoss(rand(10,15))
+	else
+		if(user)
+			user.pixel_x = user.base_pixel_x
+			user.pixel_y = user.base_pixel_y
+		if(target)
+			target.pixel_x = target.base_pixel_x
+			target.pixel_y = target.base_pixel_y
+
+	log_combat(user, target, "body-slammed (Wrestling)")
+	return
+
+/datum/martial_art/fulpwrestling/proc/FlipAnimation(mob/living/D)
+	set waitfor = FALSE
+	if(D)
+		animate(D, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+	if(do_after(D, 10 SECONDS))
+		if(D)
+			animate(D, transform = null, time = 1, loop = 0)
+
+/// Drop
+/datum/martial_art/fulpwrestling/proc/fulpdrop(mob/living/user, mob/living/target)
```

Comment:
ironically, it's most optimal to use this move when NOT adjacent to a table, as that removes half of the windup period of the move (and your target can't dodge the move by just running away during the windup period that not being adjacent to a table keeps, so it's really just a delay), makes it impossible for the move to backfire, and allows you to damage conscious targets.